Rater.Business/Services/SpaceService.cs

```diff
@@ -12,25 +16,48 @@ namespace Rater.Business.Services
     public class SpaceService : ISpaceService
     {
 
-        private readonly ISpaceRepository _repo;
-        public SpaceService(ISpaceRepository repo)
+        private readonly ISpaceRepository _spaceRepo;
+        private readonly IUserRepository _userRepo;
+        private readonly IMapper _mapper;
+        public SpaceService(ISpaceRepository spaceRepo,IUserRepository userRepo,IMapper mapper)
         {
-            _repo = repo;
+            _spaceRepo = spaceRepo;
+            _userRepo = userRepo;
+            _mapper = mapper;
         }
 
 
 
         public async Task<List<SpaceResponseDto>> GetAllSpaces()
         {
-            var value = await _repo.GetAllSpaces();
+            var value = await _spaceRepo.GetAllSpaces();
             return value;
         }
 
-        public async Task<SpaceResponseDto> CreateSpace(SpaceRequestDto request)
+        public async Task<SpaceResponseDto> AddSpace(GrandSpaceRequestDto request)
         {
-            var value = await _repo.CreateSpace(request);
-            return value;
+            UserRequestDto userRequest = new UserRequestDto();
+            userRequest.NickName = request.creatorNickname;
+            var justCreatedUser = await _userRepo.AddUser(userRequest);
+
+
+            var space = _mapper.Map<Space>(request);
+            space.CreatorId = await _userRepo.GetCreatorId(justCreatedUser.CreatedAt);
```

Comment:
F1 - Why using a method to get userId, can't you use `justCreatedUser` object's user id?

Reply:
After we create a user and insert it into our table, we no longer have access to the exact object with its ID information, as the ID is automatically assigned by the database. Therefore, we need a method to retrieve the ID to set the creator for newly created spaces.

We could also consider different approaches, such as fetching the last object from the database, but this might introduce bugs in large-scale projects.

Reply:
I just reviewed my code and made some changes.